src/os/bluestore/BlueFS.cc

```diff
@@ -1941,7 +1942,8 @@ int64_t BlueFS::_read_random(
       logger->inc(l_bluefs_read_random_disk_count, 1);
       logger->inc(l_bluefs_read_random_disk_bytes, l);
       if (len > 0) {
-	s_lock.lock();
+        u_lock.unlock();
+ 	 s_lock.lock();
```

Comment:
I'm unclear how this protects ```h->buf->bl``` -- it looks like it's only modified in the else branch below

Reply:
it looks like line 1958's memcpy is the only place we might be allocating new memory for buf->bl - is there a way to put it in the correct mempool directly with the c_str() call it results in new allocations?

Reply:
@aclamk ^

Reply:
@jdurgin, reassigning buf when it is in locked state, resolve the thrash failures: http://pulpito.front.sepia.ceph.com/ideepika-2020-06-11_11:55:48-rados:thrash-erasure-code-isa-test-bluestore-mem-tracking-distro-basic-smithi/
I am not sure which one[out/h->buf->bl] would account for correct mem values, testing that locally.

Reply:
@jdurgin I think that c_str() clearing assignment to mempool is a bug. Will make PR to fix that.